frontend/src/modules/SimulationTimeSeriesMatrix/settings.tsx

```diff
@@ -81,6 +76,36 @@ export function settings({ moduleContext, workbenchSession }: ModuleFCProps<Stat
         setPreviousEnsembleSet(ensembleSet);
     }
 
+    // Queries
+    const vectorListQueries = useVectorListQueries(selectedEnsembleIdents);
+    const ensembleVectorListsHelper = new EnsembleVectorListsHelper(selectedEnsembleIdents, vectorListQueries);
+    const selectedVectorNamesHasHistorical = ensembleVectorListsHelper.hasAnyHistoricalVector(selectedVectorNames);
+    const currentVectorSelectorData = createVectorSelectorDataFromVectors(ensembleVectorListsHelper.vectorsUnion());
+
+    // Get union of continuous and non-constant parameters for selected ensembles and set valid parameter ident str
+    const continuousAndNonConstantParametersUnion: ParameterIdent[] = [];
+    for (const ensembleIdent of selectedEnsembleIdents) {
+        const ensemble = ensembleSet.findEnsemble(ensembleIdent);
+        if (ensemble === null) continue;
```

Comment:
As the `ensembleIdent` is retrieved from the `ensembleSet` I mean this is not the responsibility of the module settings. This check is more to make typescript happy as `findEnsemble` returns null rather than throwing error.

If it was to be fixed, I'd rather enrich the `EnsembleSet` class with an `hasEnsemble` method to check for specific `ensembleIdent` and a `getEnsemble` which throws error if asking for non-existing ensemble.